[SRVOCF-442] Move on-cluster function building to separate assembly; add the section on specifying which function to build (#51183)
openshift/openshift-docs: modules/serverless-functions-specifying-function-revision.adoc

// Module included in the following assemblies:
//
// * serverless/functions/serverless-functions-on-cluster-builds.adoc

:_content-type: PROCEDURE
[id="serverless-functions-specifying-function-revision_{context}"]
= Specifying function revision

> When building and deploying a function on the cluster as opposed to locally, you need to point to the correct function revision by specifying the Git repository, branch, location within the repository, and other parameters. In addition to specifying these parameters using the `func.yaml` configuration file, you can use the flags for the `kn func deploy` command.

Reviewer: ```suggestion
When building and deploying a function on the cluster, you must point to the correct function revision, by specifying the Git repository, branch, and location within the repository. You can specify these parameters in the `func.yaml` configuration file, or by using flags with the `kn func deploy` command.
```

Reviewer 2: I agree with these suggestions, but also, I would make some additional changes.
```suggestion
When building and deploying a function on the cluster as opposed to locally, you need must specify the location of your code. This includes the Git repository URL, the branch if not the main branch, and any subdirectory within the repository if your function is not at the root level. You can specify these parameters in the `func.yaml` configuration file, or by using flags with the `kn func deploy` command.
```
The meaning is just slightly different. You don't specify a "revision", but rather a location. Additionally, the subdirectory and branch are not required.

.Prerequisites

* {pipelines-title} must be installed on your cluster.

* You have installed the OpenShift CLI (`oc`).

> * You have installed the Knative (`kn`) CLI.

Reviewer: Make these consistent, i.e. brackets either before or after "CLI" in both cases. I think for our other docs it's before.

.Procedure

* Deploy your function:
+
[source,terminal]
----
$ kn func deploy --remote \

>                  --build git \

Reviewer 3: This `--build git` is not required anymore. Please remove it.
```suggestion
```

		 --git-url <repo-url> \

>                  --git-branch <branch> \

Reviewer 2: Can these flags line up?

Author: @lance Line up meaning go into a single line? I'd rather not:
* Per Ashleigh's suggestion, the options now have numbered callouts, which work better with multi-line listing. (I don't think they work in-line at all, in fact)
* The example at the end contains a non-short URL, so IMO it's better to split & align the individual flags.

Feel free to push back again though & I'll change it to a single line. FWIW though, copypasting this example works nicely in ksh and bash and probably most other shells.

Reviewer 2: I actually meant vertical alignment like this.

```
$ kn func deploy --remote \
                 --git-url <repo-url> \
                 --git-branch <branch>
```
Also, note here you don't need the `--build git` flag.

Author: Thanks! `--build git` removed and tabs -> spaces fixed. :man_facepalming:

		 --git-dir <function-dir>
----
+
--
** With the `--remote` flag, the build runs remotely.
** Substitute `<repo-url>` with the URL of the Git repository.

> ** Substitute `<branch>` with the Git branch, tag, or commit.

Reviewer 2: This should be noted as optional if the function is on the main branch of the repo.

> ** Substitute `<function-dir>` with the directory containing the function if it is different than the repository root directory.

Reviewer: This seems like it could use proper callouts, e.g. `<1>`?

--
+
For example:
+
[source,terminal]
----
$ kn func deploy --remote \
		 --git-url https://example.com/alice/myfunc.git \

>                  --git-branch my-feature \

Reviewer 2: Same here - can these be aligned?

Author: See the above explanation, but again feel free to re-request this of me.
